auth/manager_test.go

```diff
@@ -0,0 +1,171 @@
+package auth
+
+import (
+	"net/http"
+	"testing"
+	"time"
+)
+
+const jwt = "eyJraWQiOiJzTjVnTk43cWFGVmpPYVwvc3oyVTJvdnNIMTZyNThQb2RQVFpRZWlBQUhNZz0iLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJlMjY1YmNhMS03NGZiLTQ2MjQtOWVlMC0zOGE5ZmQ5YTQ4OTUiLCJldmVudF9pZCI6Ijk4MjUxMzQ3LTFmN2ItNGY0OC1hNTNhLWE3MjZhOWEzOTFiNiIsInRva2VuX3VzZSI6ImFjY2VzcyIsInNjb3BlIjoiYXdzLmNvZ25pdG8uc2lnbmluLnVzZXIuYWRtaW4iLCJhdXRoX3RpbWUiOjE2MjM0MDYwODQsImlzcyI6Imh0dHBzOlwvXC9jb2duaXRvLWlkcC5ldS1jZW50cmFsLTEuYW1hem9uYXdzLmNvbVwvZXUtY2VudHJhbC0xX1ZhSGgwRW9YNSIsImV4cCI6MTYyMzQyNzY4NCwiaWF0IjoxNjIzNDA2MDg0LCJqdGkiOiIxOWUwMTZmOS05NGU0LTQ1ZTYtYmE4OS0xYjg4Y2ZmMThmN2QiLCJjbGllbnRfaWQiOiI0aW8zYWVxZjFoOTY3dWZhbGs1bjc0MmNmaiIsInVzZXJuYW1lIjoibXdleWEifQ.I5Nn5selXHZVPIJFJ0HSAoqUtQxz9s37e-6YLua2S0M9xcNhx0h-Yr6A3S8JNH4OXsCKYK3r-Y9TOjce4CGRQRTprDIKQnwS3RMOz1jTT6YVXy5n8CEvN2ySC8y-27oLLOjAHLGQjRaj_o8PkrQVSzSxKJbthTS0fNodPTsBt6FpXEZv5ULIJzFnWVxjPD3Rb0h6Ts-iYylvcurWiCKZYQLItuydoCG_99uuIlL1BzDHxBz-QRJYRF_gFJlQbUBfsXHv3VT_ET5vTkXeNILqdq_ON2PrcbXR1uDXm2wvF4TKZsD3tFRFEeH6FK6xM0dnMnogphEowZyU_BIEQlgTsw"
+
+func TestCreateManager(t *testing.T) {
+	_ = NewManager()
+	t.Logf("Manager created!")
+}
+
+func TestCreateAndWriteToManager(t *testing.T) {
+	manager := NewManager()
+	manager.bearerAuth = false
+	manager.srv = &http.Server{}
+	manager.tokenChan = make(chan *Token)
+	manager.errChan = make(chan error, 1)
+	t.Logf("Manager fields modified!")
+}
+
+func TestManagerStoreTokens(t *testing.T) {
+	manager := NewManager()
+	token := Token{
+		AccessToken:     jwt,
+		IDToken:         "idToken",
+		RefreshToken:    "refreshToken",
+		Expires:         0,
+		DetaAccessToken: "aaaaaaaaaaaaaaaaaaaaaaaaaa",
+	}
+	manager.storeTokens(&token)
+	t.Logf("Token stored by manager!")
+}
+
+func TestManagerGetToken(t *testing.T) {
+	manager := NewManager()
+	_, err := manager.getTokens()
+	if (err != nil) {
+		t.Errorf("Error retrieving token: %s", err.Error())
+	}
+	t.Logf("Token successfully retrieved")
+}
+
+func TestManagerGetTokenAndRead(t *testing.T) {
+	manager := NewManager()
+	token, err := manager.getTokens()
```

Comment:
how do you retrieve the token without storing it first? Each test case should be able to be run on it's own.

Reply:
meaning it should not rely on other test cases for `set up` or `teardown`

Reply:
Yeah, I was depending on go test running the one above before the one below, I'll redo that